zanata-war/src/main/java/org/zanata/service/impl/TranslationUpdatedManager.java

```diff
@@ -55,23 +61,28 @@ public void textFlowStateUpdated(
     // Fire asynchronous event
     void publishAsyncEvent(TextFlowTargetStateEvent event) {
         if (BeanManagerProvider.isActive()) {
-            int wordCount = textFlowDAO.getWordCount(event.getTextFlowId());
+            Long versionId = event.getKey().getProjectIterationId();
+            Long documentId = event.getKey().getDocumentId();
+            LocaleId localeId = event.getKey().getLocaleId();
 
-            // TODO use Event.fire()
-            BeanManagerProvider.getInstance().getBeanManager().fireEvent(
-                    new DocumentStatisticUpdatedEvent(
-                            event.getProjectIterationId(),
-                            event.getDocumentId(), event.getLocaleId(),
-                            wordCount,
-                            event.getPreviousState(), event.getNewState())
-            );
+            for(TextFlowTargetState state: event.getStates()) {
+                int wordCount =
+                    textFlowDAO.getWordCount(state.getTextFlowId());
+                documentStatisticUpdatedEvent
+                    .fire(new DocumentStatisticUpdatedEvent(
```

Comment:
We should be able to reduce the number of DocumentStatisticUpdatedEvents now that we have aggregation. We should put a todo here, perhaps something like:

    // TODO turn DocumentStatisticUpdatedEvent into an aggregate event like TextFlowTargetStateEvent
    // TODO generate one DocumentStatisticUpdatedEvent per TextFlowTargetStateEvent